File: frontend/src/api/health.ts
// ABOUTME: Health check API endpoints for monitoring backend status
// ABOUTME: Provides health and readiness check functionality

import type { HealthResponse, ReadinessResponse } from '../types/api'
import apiClient from './client'

export const healthApi = {
  // Basic health check
  getHealth: (): Promise<HealthResponse> => {
    return apiClient.get<HealthResponse>('/health')
  },

  // Readiness check (includes database connectivity)
  getReadiness: (): Promise<ReadinessResponse> => {
    return apiClient.get<ReadinessResponse>('/ready')
  },

  // Root endpoint info
  getInfo: (): Promise<{
    name: string
    version: string
    status: string
    environment: string
  }> => {
    return apiClient.get('/')
  },
}

File: backend/app/api/v1/health.py
# ABOUTME: Health and readiness check endpoints for monitoring
# ABOUTME: Provides basic health status and database connectivity checks

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import text
from sqlalchemy.orm import Session

from app.core.config import config
from app.db.database import get_db

router = APIRouter(tags=["health"])


@router.get("/health")
async def health_check():
    """Basic health check endpoint"""
    return {
        "status": "healthy",
        "service": "foosball-api",
        "version": config.version,
        "environment": config.environment,
    }


@router.get("/ready")
async def readiness_check(db: Session = Depends(get_db)):
    """Detailed readiness check including database connectivity"""
    try:
        # Test database connection
        db.execute(text("SELECT 1"))

        return {
            "status": "ready",
            "service": "foosball-api",
            "version": config.version,
            "environment": config.environment,
            "checks": {"database": "healthy"},
        }
    except Exception as e:
        raise HTTPException(
            status_code=503,
            detail={
                "status": "not_ready",
                "error": str(e),
                "checks": {"database": "unhealthy"},
            },
        )
